### PR TITLE
Display chunk report in hex values

### PR DESCRIPTION
**Pull request by aravind-wdc**
Small change.
As per Matias's suggestion, changed the command to show in hex values.

### CODE REVIEW

**Comment by MatiasBjorling** [comment]
The thinking was to keep nlb and wp in hex as well, since they are all addresses or count of LBAs

**Comment by safl** [comment]
@aravind-wdc and @MatiasBjorling thanks for the PR, however, is this just a cosmetic change due to personal preference for hex over dec? Or, is there more to it?

**Comment by MatiasBjorling** [comment]
Usually, the library uses hex for addresses, and it is also the way chunks are exposed through blkzone and zbd_report_zones tools. So I wanted to keep it consistent.

**Comment by aravind-wdc** [comment]
Hi Simon,

I got a bit confused (before seeing the code) at the output of rprt_all, so was checking with Matias whether my understanding was correct, so at that point we thought making all values in hex may alleviate the confusion. That's all.

Regarding the second change,  271765148555657d6d68f10f55335244cebb3a07, the lba2gen option is returning incorrect value (0) , so fixed it. Kindly review.

Adding one more change and to the same PR.

"nvm_addr lba2gen " always returns 0.

=================================
aravind@ocssd:~/OCSSD/liblightnvm/liblightnvm/build/cli$ sudo ./nvm_addr gen2lba /dev/nvme0n1 0x0000000100000003
gen: 0x0000000100000003, lba: 0000000000000000000000000000000000000000000000000000000000032792
aravind@ocssd:~/OCSSD/liblightnvm/liblightnvm/build/cli$ sudo ./nvm_addr lba2gen /dev/nvme0n1 0000000000000000000000000000000000000000000000000000000000032792
lba: 0000000000000000000000000000000000000000000000000000000000032792
gen: naddrs: 1
addrs:
  - {val: 0x0000000000000000, pugrp: 00, punit: 00, chunk: 0000, sectr: 0000}
aravind@ocssd:~/OCSSD/liblightnvm/liblightnvm/build/cli$

============With the fix output below==========================
aravind@ocssd:~/OCSSD/liblightnvm/fix_lba/liblightnvm/build/cli$ sudo ./nvm_addr gen2lba /dev/nvme0n1 0x0000000100000003 0x0000000100000004 0x0000000100000005
gen: 0x0000000100000003, lba: 0000000000000000000000000000000000000000000000000000000000032792
gen: 0x0000000100000004, lba: 0000000000000000000000000000000000000000000000000000000000032800
gen: 0x0000000100000005, lba: 0000000000000000000000000000000000000000000000000000000000032808
aravind@ocssd:~/OCSSD/liblightnvm/fix_lba/liblightnvm/build/cli$
aravind@ocssd:~/OCSSD/liblightnvm/fix_lba/liblightnvm/build/cli$
aravind@ocssd:~/OCSSD/liblightnvm/fix_lba/liblightnvm/build/cli$ sudo ./nvm_addr lba2gen /dev/nvme0n1 0000000000000000000000000000000000000000000000000000000000032792 0000000000000000000000000000000000000000000000000000000000032800 0000000000000000000000000000000000000000000000000000000000032808
lba: 0000000000000000000000000000000000000000000000000000000000032792
gen: naddrs: 1
addrs:
  - {val: 0x0000000100000003, pugrp: 00, punit: 00, chunk: 0001, sectr: 0003}
lba: 0000000000000000000000000000000000000000000000000000000000032800
gen: naddrs: 1
addrs:
  - {val: 0x0000000100000004, pugrp: 00, punit: 00, chunk: 0001, sectr: 0004}
lba: 0000000000000000000000000000000000000000000000000000000000032808
gen: naddrs: 1
addrs:
  - {val: 0x0000000100000005, pugrp: 00, punit: 00, chunk: 0001, sectr: 0005}
aravind@ocssd:~/OCSSD/liblightnvm/fix_lba/liblightnvm/build/cli$


=================================================

**Comment by safl** [comment]
Yes, liblightnvm mostly uses hex for addresses/enum-types and decimal for counts/values. By that argument only WP should be changed, don't you agree?
Aligning with other tools and reducing the chance of confusion seems like better motivation, as it makes cross-referencing output by humans more convenient. However, there is just one concern in this regard, which is that there are users with brittle non-human consumption of the output, brittle because the value-parsing does not cater to the "0x" convention, so the representation is hardcoded in the parsing of the values, which in turn breaks tests/scripts/tools when changing from decimal to hexi-decimal. Anyway, I'll postpone pulling that change in, giving time to fix their parsing.

**TLDR** I'll wait with the nvm_spec_rprt_pr change. The change to the lba2gen CLI is good, thanks!

**Comment by MatiasBjorling** [comment]
Well, generally, the parsing of the output is harder than it needs to it for computers. This should properly be fixed. I'm proposing to add a terse argument that will output the stuff in better machine readable format.

For WP, well, if WP is hex, seeing the nlb in non-hex next to it is more confusing.  E.g., 4096  is 0x1000 in hex. I much rather see nlb 0x1000, so I know that the WP is correct.

**Comment by MatiasBjorling** [comment]
Also, assuming that we will have more of this "breaking" changes, let's make sure to put new/changes to outputs up for PR, so we have a chance to chime in before it goes into master.

**Comment by safl** [comment]
Having a CLI that outputs in a standardized structured format does not need "fixing" since parsing the output is not hard for computers, it is trivial.

An example of something that does need fixing in this area is the output of the OCSSD 1.2 CLI nvm_bbt which is in a custom pseudo-structured and not trivially parsed format.

**Comment by safl** [comment]
@aravind-wdc I've added the hex/dec change to the **pprint** branch, if you have more of the sort, please set up PR towards it. The fix to the CLI **nvm_addr lba2gen** has gone into master, thanks.

**Comment by MatiasBjorling** [comment]
I maybe wasn't clear. What I meant, is that it is parse something like this:

 ./nvm_addr lba2gen /dev/nvme0n1 0000000000000000000000000000000000000000000000000000000000032792
lba: 0000000000000000000000000000000000000000000000000000000000032792
gen: naddrs: 1
addrs:

{val: 0x0000000000000000, pugrp: 00, punit: 00, chunk: 0000, sectr: 0000}

Instead of just exposing the LBA only. The other is fluff, which I then need to build up a regex for or something else. Instead of just being able to pipe from one exec to the other.

**Comment by safl** [comment]
I agree, in that use-case, it would be convenient with a "-t" flag, or the like, for the **nvm_addr** CLI, such that only the value of the address conversion would be output and conveniently piped to another process. That would be a welcome feature.

**Comment by MatiasBjorling** [comment]
What about the other commit? Should we keep this open until it is upstream?

If you have infrastructure that depends on this, then it would be a good time make sure that infrastructure only uses the latest stable, and we can use the version bump when there are updates to visible APIs.

**Comment by MatiasBjorling** [comment]
Such that development is not slowed down because of internal churn

**Comment by MatiasBjorling** [comment]
Opening so we don't forget.
